### PR TITLE
[IMPROVE] Deal with new backend state on email validation error

### PR DESCRIPTION
**Pull request by MartinSchoeler**
**Will be ready to review as soon as https://github.com/RocketChat/Rocket.Chat/pull/25312 is merged**

### CODE REVIEW

File: packages/livechat/src/lib/hooks.js [hunk 1]
```diff
@@ -137,7 +137,9 @@ const api = {
 			api.setDepartment(data.department);
 		}
 
-		createOrUpdateGuest(data);
+		createOrUpdateGuest(data).catch((e) => {
```
Comment (MartinSchoeler) [inline packages/livechat/src/lib/hooks.js:140]: Screen Shot 2022-05-05 at 10 43 13 AM
@KevLehman

File: packages/livechat/src/routes/Chat/container.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ class ChatContainer extends Component {
 		}
 
 		const visitor = { token, ...guest };
-		const newUser = await Livechat.grantVisitor({ visitor });
+		const newUser = await Livechat.grantVisitor({ visitor }).catch(() => null);
```
Comment (MartinSchoeler) [inline packages/livechat/src/routes/Chat/container.js:60]: Screen Shot 2022-05-05 at 10 43 56 AM
@KevLehman

File: packages/livechat/src/routes/Register/container.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ export class RegisterContainer extends Component {
 
 		await dispatch({ loading: true, department });
 		try {
-			const user = await Livechat.grantVisitor({ visitor: { ...fields, token } });
+			const user = await Livechat.grantVisitor({ visitor: { ...fields, token } }).catch(() => null);
```
Comment (MartinSchoeler) [inline packages/livechat/src/routes/Register/container.js:44]: Screen Shot 2022-05-05 at 10 44 36 AM
@KevLehman

File: packages/livechat/src/routes/SwitchDepartment/container.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ class SwitchDepartmentContainer extends Component {
 		}
 
 		if (!room) {
-			const user = await Livechat.grantVisitor({ visitor: { department, token } });
+			const user = await Livechat.grantVisitor({ visitor: { department, token } }).catch(() => null);
```
Comment (MartinSchoeler) [inline packages/livechat/src/routes/SwitchDepartment/container.js:31]: Screen Shot 2022-05-05 at 10 45 03 AM
@KevLehman

**Review by KevLehman — DISMISSED** [review]
My prev comments are no longer valid :) i'll approve, but @tiagoevanp , go ahead and review it (it requires a FE review anyways lol)

**Comment by KevLehman** [comment]
This has a conflict 👀

File: packages/livechat/src/routes/SwitchDepartment/container.js [hunk 5]
```diff
@@ -28,7 +28,12 @@ class SwitchDepartmentContainer extends Component {
 		}
 
 		if (!room) {
-			const user = await Livechat.grantVisitor({ visitor: { department, token } });
+			let user = null;
+			try {
+				user = await Livechat.grantVisitor({ visitor: { department, token } });
+			} catch (e) {
+				console.error(e);
+			}
```
Comment (ggazzo) [inline packages/livechat/src/routes/SwitchDepartment/container.js:36]:
```suggestion
			const user =  await Livechat.grantVisitor({ visitor: { department, token } }).catch ((e) => {
				console.error(e);
				return Promise.reject(e); // not sure if should fail or return undefined
			});
```

**Pull request closed without merging**